File: old_projects/src/project2/utility/Heuristics.java
package project2.utility;

public class Heuristics {
    /**
     * This program implements the euclidean distance calculation for heuristic cost
     *
     * @param currentCoordinate Current Coordinate
     * @param goalCoordinate Goal Coordinate
     * @return Returns a double value
     */
    public static double euclideanDistance(Point currentCoordinate, Point goalCoordinate) {
        double tempX = Math.pow(currentCoordinate.f1 - goalCoordinate.f1, 2);
        double tempY = Math.pow(currentCoordinate.f2 - goalCoordinate.f2, 2);

        return Math.sqrt(tempX + tempY);
    }

    /**
     * This program implements the manhattan distance calculation for heuristic cost
     *
     * @param currentCoordinate Current Coordinate
     * @param goalCoordinate Goal Coordinate
     * @return Returns a double value
     */
    public static double manhattanDistance(Point currentCoordinate, Point goalCoordinate) {
        double tempX = Math.abs(currentCoordinate.f1 - goalCoordinate.f1);
        double tempY = Math.abs(currentCoordinate.f2 - goalCoordinate.f2);

        return tempX + tempY;
    }

    /**
     * This program implements the chebyshev distance calculation for heuristic cost
     *
     * @param currentCoordinate Current Coordinate
     * @param goalCoordinate Goal Coordinate
     * @return Returns a double value
     */
    public static double chebyshevDistance(Point currentCoordinate, Point goalCoordinate) {
        double tempX = Math.abs(currentCoordinate.f1 - goalCoordinate.f1);
        double tempY = Math.abs(currentCoordinate.f2 - goalCoordinate.f2);

        return Math.max(tempX, tempY);
    }

}

File: Simulator/Utility/Heuristics.py
from Utility.Point import Point


class Heuristics:
    @staticmethod
    def euclideanDistance(currentCoordinate: Point, goalCoordinate: Point):

        return ((currentCoordinate.f2 - goalCoordinate.f2)**2 +
                (currentCoordinate.f1 - goalCoordinate.f1)**2)**0.5

    @staticmethod
    def manhattanDistance(currentCoordinate: Point, goalCoordinate: Point):
        return ((abs(currentCoordinate.f1 - goalCoordinate.f1) +
                 abs(currentCoordinate.f2 - goalCoordinate.f2)))

    @staticmethod
    def chebyshevDistance(currentCoordinate: Point, goalCoordinate: Point):
        return max(abs(currentCoordinate.f1 - goalCoordinate.f1),
                   abs(currentCoordinate.f2 - goalCoordinate.f2))


if __name__ == '__main__':
    print(Heuristics.manhattanDistance(Point(0, 0), Point(1, 1)))
